### PR TITLE
Enabled ',' (comma) in StringWithMacro expression

### PR DESCRIPTION
**Pull request by airween**
This is another form of #36.

Enabled `,` (comma) character in some cases (eg. by `logdata`). With this, the below form is allowed:

```
logdata:'blocked foo, param_with_underscore=%{TX:1}, value_with_underscore=%{TX:2}'
```
Perhaps we have to review the allowed arguments type at the actions, I think there are more actions which allow this in the engine(s) than here. And there is an another used solution, eg.
```
   125      'msg:' msg=MessageValue | ...
   ...
   216  MessageValue: "'" /((\\')|[^\'])*/ "'";
```
which - may be - we can use.

### CODE REVIEW

File: src/secrules_parsing/model/secrules.tx [hunk 1]
```diff
@@ -197,7 +197,10 @@ Macro:  INT | OperationMacro | ExtendedMacro | MacroVar;
 
 // This extended macro are a little bag of holding...
 ExtendedMacro: /[A-Za-z0-9S\-\_\.\:\ \/\%\{\}\|\+=#?]+/;
-    
+
+// macro with comma, quoted by '
+ExtendedMacroWithComma: /[A-Za-z0-9S\-\_\.\:\ \/\%\{\}\|\+=#?,]+/;
```
Comment (theseion) [inline src/secrules_parsing/model/secrules.tx:202]: Maybe a dumb question: why does the regex have to be so complex? Can't it just be (approximately) `"[^"]+"|'[^']+'`?

Comment (airween) [reply src/secrules_parsing/model/secrules.tx:202]: It's not a dumb question, it's a valid question.

I haven't investigated it, just allowed what there is. :smile:

> Can't it just be any character within quotes and anything but comma without quotes?

Perhaps that can be. See my related comment: _we have to review the allowed arguments type at the actions..._

Note, that libmodsecurity3 uses [this](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/src/parser/seclang-scanner.ll#L434) pattern.

Comment (theseion) [reply src/secrules_parsing/model/secrules.tx:202]: They use that to find macros within the string. But I can see that we might want to parse the macro one day....

**Review by theseion — APPROVED** [review]

File: src/secrules_parsing/model/secrules.tx (hunk 1 above)
Comment (fzipi) [reply src/secrules_parsing/model/secrules.tx:202]: I have a branch to decouple things a bit and make things more semantic in the future. But not for now.

**Review by fzipi — APPROVED** [review]

**Pull request merged**